xbmc/platform/win32/WinMain.cpp

```diff
@@ -88,6 +91,55 @@ static std::shared_ptr<CAppParams> ParseCommandLine()
   return appParamParser.GetAppParams();
 }
 
+/*!
+ * \brief Detect another running instance using the same data directory, using a flag file exclusion
+ * mechanism.
+ * 
+ * Attempt to create the file. If it already exists, another instance using the same data
+ * directory is running. New in v22.
+ * 
+ * \param[in] usePlatformDirectories false for portable mode, true for non-portable mode
+ * \param[out] handle Windows handle for the flag file.  Close the handle on program exit, not earlier.
+ * \return true if the flag file already exists (ie another instance is running)
+ */
+bool CheckAndSetFileFlag(bool usePlatformDirectories, HANDLE& handle)
+{
+  const std::string file = CWIN32Util::GetProfilePath(usePlatformDirectories) + "\\.running";
+
+  constexpr DWORD NO_DESIRED_ACCESS = 0;
+  constexpr DWORD NO_SHARING = 0;
+
+  using KODI::PLATFORM::WINDOWS::ToW;
+  handle = CreateFile(ToW(file).c_str(), NO_DESIRED_ACCESS, NO_SHARING, nullptr, CREATE_NEW,
+                      FILE_FLAG_DELETE_ON_CLOSE, NULL);
+
+  if (handle == INVALID_HANDLE_VALUE && GetLastError() == ERROR_FILE_EXISTS)
+    return true;
+
+  return false;
+}
+
+/*!
+ * \brief Detect another running instance, using a global mutex.
+ * 
+ * Attempt to create the mutex. If it already exists, that means another instance created it and is
+ * running. This is the legacy pre-v22 exclusion mechanism.
+ * 
+ * \param[out] handle Windows handle for the mutex. Close the handle on program exit, not earlier.
+ * \return true if the mutex already exists (ie another instance is running)
+ */
+bool CheckAndSetMutex(HANDLE& handle)
+{
+  using KODI::PLATFORM::WINDOWS::ToW;
+  handle = CreateMutex(nullptr, FALSE,
+                       ToW(std::string(CCompileInfo::GetAppName()) + " Media Center").c_str());
```

Comment:
This code is not much readable and even confuse because final string variable is not used and CreateMutex parameter is only wchar_t*.   Why not only:

```c++
handle = CreateMutexW(nullptr, FALSE, L"Kodi Media Center");
```

After all the name of mutex is not very important because only is used here (even if in a future Kodi changes the name). Also call `CreateMutexW` to denote the intentionality of use wide char version.